### PR TITLE
Default npm algorithm

### LINKED ISSUE #700: Add npmAlgorithm: nested
Since flat will become the default in 1.0.

### PR DESCRIPTION
**Pull request by pYr0x**
#close #700 #701 
@matthewp let me know about your thoughts.. i placed a note in https://github.com/pYr0x/steal/blob/default-npm-algorithm/test/default-npm-algorithm/npm-algorithm.js#L4-L8

what do you think? should we provide a configuration about `npmAlgorithm` in stealconfig or should `npmAlgorithm` always configured in package.json because of this lines https://github.com/pYr0x/steal/blob/default-npm-algorithm/ext/npm.js#L54-L56

### CODE REVIEW

File: test/default-npm-algorithm/default.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+System.import("some").then(function (module) {
+	if (typeof window !== "undefined" && window.QUnit) {
+		QUnit.strictEqual(System.npmContext.isFlatFileStructure, true, "default npm-algorithm is flat");
+		// you cant config npmAlgorithm outside of a package.json config-file!
+		// npmAlgorithm is "nested", but "System.npmContext.isFlatFileStructure" is true because of the
+		// package-config. it is a wired to have a test like this, but we should figure out if this behaviour
+		// is wanted.
+		QUnit.equal(System.npmAlgorithm, 'nested');
```
Comment (pYr0x) [inline test/default-npm-algorithm/default.js]: @matthewp what do you think? see also the notes in `default.html` above

**Comment by pYr0x** [comment]
can't get system-npm working with steal 0.16 and tests are still not passing.
with an updated system-npm this PR should pass

**Comment by pYr0x** [comment]
with this PR in steal/system-npm https://github.com/stealjs/system-npm/pull/146 the test should be pass.

@matthewp pls let me know if all is alright. also pls look at my fist comment with the note i passed.

**Comment by matthewp** [comment]
Can you also update the markdown file here: https://github.com/stealjs/steal/blob/master/docs/module-npm.md#packagesystemnpmalgorithm  Thanks!

**Comment by pYr0x** [comment]
of course...

**Comment by matthewp** [comment]
So, not sure how to merge these in. I don't want to merge them into master yet, because release is still likely a few weeks out.  We could make a **major** branch and merge 1.0 stuff into there.  Or just leave the PRs open until we are closer to having everything ready. Any thoughts?

**Comment by pYr0x** [comment]
yep... i think merge that into a major branch would be good. in that branch we can track all the 1.0 changes.
in the major branch we have to point to this system-npm branch https://github.com/stealjs/system-npm/tree/default-npm-algorithm
or create also a major branch for system-npm

**Comment by matthewp** [comment]
I think we create a major branch for system-npm and I'll do a 1.0.0-pre there as well.

**Comment by pYr0x** [comment]
ok if you are finsh with creating branches let me know and i will make a PR

**Comment by matthewp** [comment]
I'm created major branches here and in system-npm

**Comment by pYr0x** [comment]
close this.
i will create a new PR against major

**Pull request closed without merging**